src/Eshopworld.WorkerProcess/Infrastructure/SystemTimer.cs

```diff
@@ -10,44 +11,43 @@ namespace EShopworld.WorkerProcess.Infrastructure
     [ExcludeFromCodeCoverage]
     public class SystemTimer : ITimer
     {
-        private readonly Timer _timer;
+        // To detect redundant calls
+        private bool _disposed = false;
 
-        public SystemTimer()
-        {
-            _timer = new Timer();
-            _timer.Elapsed += TimerElapsed;
-        }
+        private readonly CancellationTokenSource _cancellationTokenSource;
 
-        /// <inheritdoc />
-        public double Interval
+        public SystemTimer()
         {
-            get => _timer.Interval;
-            set => _timer.Interval = value;
+            _cancellationTokenSource = new CancellationTokenSource();
         }
-
         /// <inheritdoc />
-        public event EventHandler<ElapsedEventArgs> Elapsed;
-
-        /// <inheritdoc />
-        public void Start()
+        public async Task ExecuteIn(TimeSpan interval,Func<Task> executor)
         {
-            _timer.Start();
+            await Task.Delay(interval).ContinueWith(async t=>
```

Comment:
Can you pass the cancelation token to the Delay method and have it passed externally to the Execute method?